### DISCUSSION TITLE
Are there any build time benchmarks available?

### DISCUSSION THREAD

**Discussion by nikspyratos** [php/frankenphp#835] [Q&A] [answered]
Heyo.

Giving the static binary builds a go on my M1 MBP. The only change in my dockerfile is adding `pdo` and `sqlite3` extensions.

As I'm writing this it's taken 12 minutes and has not moved past the following:

<img width="1575" alt="Screenshot 2024-06-01 at 09 33 08" src="https://github.com/dunglas/frankenphp/assets/17888779/2d83b6b5-d312-446c-a8c3-bc410f22d2b4">

So not sure if it's stuck or how long the build should be expected to take, on average? I'm using Orbstack for my Docker and no CPU limit, but a 2GB RAM limit.

**Comment by withinboredom** [comment, marked as answer]
Upx takes **forever** even on native hardware. I'd recommend turning off compression if you're just building one-off binaries but turning it on to distribute (you can even run upx manually and it will have a progress bar).

**Reply by nikspyratos** [reply]
Thanks. I left for breakfast and came back, total build time 81 minutes 😅  I'll give it a go - guessing I'll have to customise the base `dunglas/frankenphp:static-builder` image.

**Reply by francoism90** [reply]
@withinboredom Is there any reason why it's the default? I building the static one now, but indeed it takes forever.

**Reply by withinboredom** [reply]
Static builds are usually production builds, where you want it optimized.
